bootspec/src/v1.rs

```diff
@@ -112,7 +112,11 @@ impl BootSpecV1 {
                 err: e,
             })?;
 
-        let kernel_file = generation.join("kernel-modules/bzImage");
+        let kernel_image_name = match system.as_str() {
+            "x86_64-linux" => "bzImage",
```

Comment:
Are there any other arch-specific image types we should handle while we're here? Apparently there's also a `zImage` type, but I don't know when that specifically gets used....

Reply:
Yeah, so here the problem is that Nixpkgs has no real way to communicate which kernel image format is used.
This should be fixed in nixpkgs top-level generation I believe and in case the file is missing, we should perform heuristic-based detection.

You have:

- zImage (compressed image)
- uImage (U-Boot images)
- bzImage (x86 specific)
- something with .efi, EFI_ZBOOT general compression format

Either case, we should move this to https://github.com/NixOS/nixpkgs/pull/244706.

Reply:
Cool, then this seems alright for now.